src/core_functions/scalar/bit/bitstring.cpp

```diff
@@ -7,12 +7,18 @@ namespace duckdb {
 //===--------------------------------------------------------------------===//
 // BitStringFunction
 //===--------------------------------------------------------------------===//
+template <bool FROM_STRING>
 static void BitStringFunction(DataChunk &args, ExpressionState &state, Vector &result) {
 	BinaryExecutor::Execute<string_t, int32_t, string_t>(
 	    args.data[0], args.data[1], result, args.size(), [&](string_t input, int32_t n) {
 		    if (n < 0) {
 			    throw InvalidInputException("The bitstring length cannot be negative");
 		    }
+		    if (!FROM_STRING) {
+			    // FIXME: this could be optimized to not go through the VARCHAR -> BIT path
+			    auto converted = Bit::ToString(input);
```

Comment:
This looks like it will not work for bitstrings that are not inlined as the `std::string` object will be destroyed. Can we add a test for this?

Reply:
Hmm that's a good point, I hadn't considered that

Reply:
I went ahead and implemented the optimization I suggested
It's mostly code stolen from `Bit::RightShift`